packages/fuels-core/src/codec.rs

```diff
@@ -43,11 +44,19 @@ mod tests {
     fn can_convert_all_from_bool_to_u64() -> Result<()> {
         let bytes: Vec<u8> = vec![0xFF; WORD_SIZE];
 
-        assert!(try_from_bytes::<bool>(&bytes)?);
-        assert_eq!(try_from_bytes::<u8>(&bytes)?, u8::MAX);
-        assert_eq!(try_from_bytes::<u16>(&bytes)?, u16::MAX);
-        assert_eq!(try_from_bytes::<u32>(&bytes)?, u32::MAX);
-        assert_eq!(try_from_bytes::<u64>(&bytes)?, u64::MAX);
+        macro_rules! test_decode {
+            ($($for_type: ident),*) => {
+                $(
+                    assert_eq!(
+                        try_from_bytes::<$for_type>(&bytes, DecoderConfig::default())?,
+                        $for_type::MAX
+                    );
+                )*
+            };
+        }
```

Comment:
This is so @segfault-magnet. Lets add 11 lines to remove 4 :smile: 

IMO we can remove one indentation level:
```suggestion
        macro_rules! test_decode {
            ($($for_type: ident),*) => {
                $(assert_eq!(
                        try_from_bytes::<$for_type>(&bytes, DecoderConfig::default())?,
                        $for_type::MAX
                );)*
            };
        }
```